### PR TITLE
[solr] Add version 7.x and 8.x support

### PR DESCRIPTION
**Pull request by supercaracal**
Hi,

We'd like to migrate Apache Solr from `5.x` to `8.x`, however, it seems several metrics keys are changed and removed since `7.x` like follows.

### Graphs

| Label | `5.x` | `6.x` | `7.x` | `8.x` |
| --- | --- | --- | --- | --- |
| Requests | Y | Y | Y | Y |
| Errors | Y | Y | Y | Y |
| Timeouts | Y | Y | Y | Y |
| AvgRequestsPerSecond | Y | Y |  |  |
| 5minRateReqsPerSecond | Y |  |  |  |
| 5minRateRequestsPerSecond |  | Y |  |  |
| 15minRateReqsPerSecond | Y |  |  |  |
| 15minRateRequestsPerSecond |  | Y |  |  |
| AvgTimePerRequest | Y | Y |  |  |
| MedianRequestTime | Y | Y |  |  |
| 75thPcRequestTime | Y | Y |  |  |
| 95thPcRequestTime | Y | Y |  |  |
| 99thPcRequestTime | Y | Y |  |  |
| 999thPcRequestTime | Y | Y |  |  |
| ClientErrors |  |  | Y | Y |
| ServerErrors |  |  | Y | Y |
| RequestTimes |  |  | Y | Y |

### `8.x` response

```
$ curl -s -S 'http://localhost:8983/solr/open_works/admin/mbeans?wt=json&stats=true&cat=QUERY&key=/select' | jq .
{
  "responseHeader": {
    "status": 0,
    "QTime": 0
  },
  "solr-mbeans": [
    "QUERY",
    {
      "/select": {
        "class": "org.apache.solr.handler.component.SearchHandler",
        "description": "Search using components: query,facet,facet_module,mlt,highlight,stats,expand,terms,debug,",
        "stats": {
          "QUERY./select.errors.count": 0,
          "QUERY./select.handlerStart": 1564978925224,
          "QUERY./select.clientErrors.count": 0,
          "QUERY./select.requestTimes.meanRate": 0.015950988833303,
          "QUERY./select.serverErrors.count": 0,
          "QUERY./select.timeouts.count": 0,
          "QUERY./select.totalTime": 9748743578,
          "QUERY./select.requests": 3123
        }
      }
    }
  ]
}
```

### `5.x` response

```
$ curl -s -S 'http://localhost:8983/solr/open_works/admin/mbeans?wt=json&stats=true&cat=QUERYHANDLER&key=/select' | jq .
{
  "responseHeader": {
    "status": 0,
    "QTime": 1
  },
  "solr-mbeans": [
    "QUERYHANDLER",
    {
      "/select": {
        "class": "org.apache.solr.handler.component.SearchHandler",
        "version": "5.2.1",
        "description": "Search using components: ",
        "src": null,
        "stats": {
          "handlerStart": 1564021845004,
          "requests": 0,
          "errors": 0,
          "timeouts": 0,
          "totalTime": 0,
          "avgRequestsPerSecond": 0,
          "5minRateReqsPerSecond": 0,
          "15minRateReqsPerSecond": 0,
          "avgTimePerRequest": 0,
          "medianRequestTime": 0,
          "75thPcRequestTime": 0,
          "95thPcRequestTime": 0,
          "99thPcRequestTime": 0,
          "999thPcRequestTime": 0
        }
      }
    }
  ]
}
```

### CODE REVIEW

**Comment by supercaracal** [comment]
```
# cd .; git clone https://go.googlesource.com/lint /home/travis/gopath/src/golang.org/x/lint
Cloning into '/home/travis/gopath/src/golang.org/x/lint'...
error: RPC failed; curl 56 GnuTLS recv error (-54): Error in the pull function.
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
package golang.org/x/lint/golint: exit status 128
```

Could you restart the build?

**Comment by susisu** [comment]
Thank you for your contribution! I will look into it.

File: mackerel-plugin-solr/lib/solr.go [hunk 1]
```diff
@@ -88,12 +98,23 @@ func (s *SolrPlugin) setStatsMbean(core string, stats map[string]interface{}, al
 					if v == nil {
 						continue
 					}
-					v2 := v.(map[string]interface{})
+					statValues := v.(map[string]interface{})
+					if s.Version >= "7.0.0" {
```
[the remark this answers is not in the document]
Comment (supercaracal) [reply mackerel-plugin-solr/lib/solr.go]: - [ ] fix version comparing

File: mackerel-plugin-solr/lib/solr.go [hunk 2]
```diff
@@ -88,12 +98,23 @@ func (s *SolrPlugin) setStatsMbean(core string, stats map[string]interface{}, al
...
+						keyConvertedStatValues := make(map[string]interface{})
+						var keyParts []string
+						for k, v := range statValues {
+							keyParts = strings.Split(k, ".")
+							if len(keyParts) >= 3 {
```
[the remark this answers is not in the document]
Comment (supercaracal) [reply mackerel-plugin-solr/lib/solr.go]: - [ ] fix magic number

[the remark this answers is not in the document]
Comment (supercaracal) [reply mackerel-plugin-solr/lib/solr.go]: fix 95a5720

File: mackerel-plugin-solr/lib/solr.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (supercaracal) [reply mackerel-plugin-solr/lib/solr.go]: fix 70132a9

**Pull request merged**